### DISCUSSION TITLE
how to provide SendHttpRequest Authorization Headers?

### DISCUSSION THREAD

**Discussion by zyphertechnologies** [elsa-workflows/elsa-core#2452] [Design & tech]
Hi, 
I am new to ELSA and so far loving it. I want to know how to send username and password for API calls using authorization headers in  SendHttpRequest activity? Please provide link to a relevant document or simply provide the java script syntax for it.

Regards,
Iftikhar

**Comment by sfmskywalker** [comment]
Hi @absolutesolutions , welcome onboard!

To set the `Authorization` header, enter a value in the **Authorization** field. For example, if you wanted to set a Basic Auth value:

![image](https://user-images.githubusercontent.com/938393/142903447-8555a01a-3282-4f22-84ad-3075bed5be4b.png)

**Important**: there is currently no `base64` function available by default. I will be adding it to the 2.4 release, but until then you can add your own JS function that encodes a string to base64.

**Reply by sfmskywalker** [reply]
I just pushed an update to `master` with two new JS functions:

`base64Encode` and `base64Decode`.
